### PR TITLE
Upgrade to 1.1.0

### PR DESCRIPTION
**Pull request by girazoki**
Upgrades dependencies to 1.1.0 version of polkadot

### CODE REVIEW

File: client/consensus/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,613 @@
+{
```
Comment (tmpolaczyk) [inline client/consensus/.vscode/launch.json:1]: Remove this file

**Comment by girazoki** [comment]
all zombienet jobs are failing because moonwall is not prepared to fetch the release from the new polkadot-sdk github repo, CC @timbrinded, for now we can test locally though

**Comment by timbrinded** [comment]
Have you tried adding the repo using the new config option in Moonwall config?

You can specify the repo author, name , binary to look for

File: container-chains/primitives/authorities-noting-inherent/src/mock.rs [hunk 2]
```diff
@@ -153,3 +153,41 @@ impl InherentDataProvider for MockAuthoritiesNotingInherentDataProvider {
         None
     }
 }
+
+impl MockAuthoritiesNotingInherentDataProvider {
```
Comment (girazoki) [inline container-chains/primitives/authorities-noting-inherent/src/mock.rs:157]: This could be reworked a bit

File: container-chains/templates/frontier/node/src/service.rs [hunk 3]
```diff
@@ -440,6 +457,7 @@ async fn start_node_impl(
         sync_service,
     };
 
+    #[allow(deprecated)]
```
Comment (girazoki) [inline container-chains/templates/frontier/node/src/service.rs:460]: add TODO to make it even clearer that we need to adapt to async backing

File: container-chains/templates/simple/node/src/command.rs [hunk 4]
```diff
@@ -333,8 +319,7 @@ pub fn run() -> Result<()> {
 				let parachain_account =
 					AccountIdConversion::<polkadot_primitives::AccountId>::into_account_truncating(&id);
 
-				let state_version = Cli::native_runtime_version(&config.chain_spec).state_version();
-				let block: Block = generate_genesis_block(&*config.chain_spec, state_version)
+				let block: Block = generate_genesis_block(&*config.chain_spec, sp_runtime::StateVersion::V1)
```
Comment (girazoki) [inline container-chains/templates/simple/node/src/command.rs:322]: Change this to reading it from runtime!

File: node/src/service.rs [hunk 5]
```diff
@@ -441,15 +444,28 @@ async fn start_node_impl(
             para_id,
             relay_chain_interface: relay_chain_interface.clone(),
             net_config,
+            sybil_resistance_level: CollatorSybilResistance::Resistant,
```
Comment (girazoki) [inline node/src/service.rs:447]: re-check

**Comment by girazoki** [comment]
modfy exportGenesisStateCommand to cumulus one in node

File: container-chains/templates/simple/node/src/command.rs (hunk 4 above)
Comment (girazoki) [reply container-chains/templates/simple/node/src/command.rs:322]: To be fair this is only used for logging purposes. I can try to change it to something where we read it from the runtime, but it would take a bit more to spin up the node. Thoughts? @tmpolaczyk @nanocryk

Comment (tmpolaczyk) [reply container-chains/templates/simple/node/src/command.rs:322]: I see, this is used to print the genesis block hash. It sounds useful, why do you say it will take more time to spin up a node?

Comment (girazoki) [reply container-chains/templates/simple/node/src/command.rs:322]: we experienced that when we changed the export-genesis-state from using a static state version value to reading it from the runtime. I dont think this should be the case if it is only done for logging purposes.

I think the best way is to print both state v0 and state v1 genesis states, and let the user read whichever is useful for it.

File: container-chains/templates/frontier/node/src/command.rs [hunk 6]
```diff
@@ -360,10 +360,14 @@ pub fn run() -> Result<()> {
 				let parachain_account =
 					AccountIdConversion::<polkadot_primitives::AccountId>::into_account_truncating(&id);
 
-                // TODO: modify
-				let block: Block = generate_genesis_block(&*config.chain_spec, sp_runtime::StateVersion::V1)
+                // We log both genesis states for reference, as fetching it from runtime would take significant timw
```
Comment (tmpolaczyk) [inline container-chains/templates/frontier/node/src/command.rs:363]: Not sure how useful is logging both, but it's amazing that generating 2 states is faster than fetching it from the runtime.

```suggestion
                // We log both genesis states for reference, as fetching it from runtime would take significant time
```

File: node/src/service.rs (hunk 5 above)
Comment (girazoki) [reply node/src/service.rs:447]: Apparently this is for consensus mechanisms where there is no sybil-resistance, unlike Aura (our case) or nimbus

File: test/moonwall.config.json [hunk 7]
```diff
@@ -60,7 +60,7 @@
                     {
                         "name": "frontier-template",
                         "binPath": "../target/release/container-chain-template-frontier-node",
-                        "options": ["--dev", "--sealing=manual"],
+                        "options": ["--dev", "--sealing=manual", "--para-id=2000"],
```
Comment (tmpolaczyk) [inline test/moonwall.config.json:63]: Why this change?

File: node/src/service.rs [hunk 8]
```diff
@@ -605,6 +625,8 @@ async fn start_node_impl(
             sync_service,
         };
 
+        // TODO: change for async backing
+        #[allow(deprecated)]
         start_full_node(params)?;
```
[the remark this answers is not in the document]
Comment (girazoki) [reply node/src/service.rs:630]: the way to start the node with async backing is different, not  with start_full_ndoe. But we will implement it in the future

File: node/src/command.rs [hunk 9]
```diff
@@ -487,8 +469,8 @@ pub fn run() -> Result<()> {
 				let parachain_account =
 					AccountIdConversion::<polkadot_primitives::AccountId>::into_account_truncating(&id);
 
-				let state_version = Cli::native_runtime_version(&config.chain_spec).state_version();
-				let block: Block = generate_genesis_block(&*config.chain_spec, state_version)
+                // TODO: fetch it from
```
[the remark this answers is not in the document]
Comment (girazoki) [reply node/src/command.rs:472]: the TODO here does not make sense, will remove it

File: container-chains/primitives/authorities-noting-inherent/src/mock.rs (hunk 2 above)
Comment (girazoki) [reply container-chains/primitives/authorities-noting-inherent/src/mock.rs:158]: yes I can do it in this one

Comment (girazoki) [reply container-chains/primitives/authorities-noting-inherent/src/mock.rs:158]: Done

File: test/suites/dev-tanssi/fees/test_fee_balance_transfer.ts [hunk 10]
```diff
@@ -54,10 +56,12 @@ describeSuite({
                 const basePlusWeightFee = (
                     await polkadotJs.call.transactionPaymentApi.queryWeightToFee(info2.weight)
                 ).toBigInt();
-                expect(basePlusWeightFee).to.equal(1000000n + 1630678n);
+                // TODO: where do these values come from?
+                expect(basePlusWeightFee).to.equal(1000000n + 1479873n);
```
Comment (tmpolaczyk) [inline test/suites/dev-tanssi/fees/test_fee_balance_transfer.ts:60]: 1000000n is the base fee, which I believe is the fee paid for a transaction with 0 weight, and the other value comes from the weight of the extrinsic and it's only here to detect when it changes. In this case it's the weight of balances.transfer, which I guess has changed with the release.

Comment (tmpolaczyk) [reply test/suites/dev-tanssi/fees/test_fee_balance_transfer.ts:60]: You can try moving the +1 here by changing the value to 1479874n.

File: test/suites/dev-tanssi/fees/test_fee_balance_transfer.ts [hunk 11]
```diff
@@ -54,10 +56,12 @@ describeSuite({
...
                 expectedBasePlusWeightFee = basePlusWeightFee;
 
-                const expectedFee = basePlusWeightFee + BigInt(signedTx.encodedLength);
+                // TODO: revisit this 1 unit that is getting injected here
+                const expectedFee = basePlusWeightFee + BigInt(signedTx.encodedLength) + 1n;
```
Comment (girazoki) [inline test/suites/dev-tanssi/fees/test_fee_balance_transfer.ts:64]: This +1 bothers me a lot

File: test/suites/dev-tanssi/fees/test_fee_balance_transfer.ts (hunk 10 above)
Comment (girazoki) [reply test/suites/dev-tanssi/fees/test_fee_balance_transfer.ts:60]: yeah but I want  to understand where it i coming from

Comment (tmpolaczyk) [reply test/suites/dev-tanssi/fees/test_fee_balance_transfer.ts:60]: Oh, you can't move it here because then this expect would fail and you would need a -1 here...

Comment (girazoki) [reply test/suites/dev-tanssi/fees/test_fee_balance_transfer.ts:60]: I think its coming from here: https://github.com/paritytech/polkadot-sdk/blob/2c7166eb9c82bedb02bba9882bc5aa5cd38b8fad/substrate/frame/transaction-payment/src/lib.rs#L761

Comment (tmpolaczyk) [reply test/suites/dev-tanssi/fees/test_fee_balance_transfer.ts:60]: I don't think so, git blame shows 2 years ago. If it's in that file, then it must be this commit https://github.com/paritytech/polkadot-sdk/commit/2c7166eb9c82bedb02bba9882bc5aa5cd38b8fad

Comment (girazoki) [reply test/suites/dev-tanssi/fees/test_fee_balance_transfer.ts]: it is a rounding error based on the fact that one api returns unadjustedFee (without multiplier) and the other returns adjusted (with multiplier). In our case we have a constant multiplier of 1, but due to rounding issues we get a +1

File: container-chains/primitives/authorities-noting-inherent/src/mock.rs [hunk 12]
```diff
@@ -60,51 +60,23 @@ pub struct MockAuthoritiesNotingInherentDataProvider {
     pub authorities: Vec<NimbusId>,
 }
 
+// TODO: lots of duplicated code between these two funcs, we should refactor
```
Comment (Agusrodri) [inline container-chains/primitives/authorities-noting-inherent/src/mock.rs:63]:
```suggestion
```

**Review by Agusrodri — APPROVED** [review]
Looks good!

File: test-sproof-builder/src/lib.rs [hunk 13]
```diff
@@ -123,24 +123,32 @@ impl ParaHeaderSproofBuilder {
...
+        // Fetch relevant keys
+        for item in self.items {
+            let para_key = item.para_id.twox_64_concat();
+            let key = [well_known_keys::PARAS_HEADS_INDEX, para_key.as_slice()].concat();
+            relevant_keys.push(key.clone());
+        }
+
         // Insert new keys and add them to relevant keys
-        {
+        /*       {
```
Comment (tmpolaczyk) [inline test-sproof-builder/src/lib.rs:146]: Should this be removed?
